pkgs/tools/typesetting/tex/texlive/bin.nix

```diff
@@ -30,6 +30,8 @@ let
       for i in texk/kpathsea/mktex*; do
         sed -i '/^mydir=/d' "$i"
       done
+
+      sed -i '/^#define ST_NLINK_TRICK/d' texk/kpathsea/config.h
```

Comment:
Excellent catch. I wonder how many other things were broken because of this, and we did not know.

Could you please add a short comment explaining why we need this patch? E.g. "ST_NLINK_TRICK causes kpathsea to treat folders with no real subfolders as leaves, even if they contain symlinks to other folders; must be disable to work correctly with the nix store" (actually if you think of anything clearer than that, that would be great).

Best if accompanied by a reference, say link to the manual if it's a stable link, or clear indication "see kpathsea manual section XXX" or similar.

Reply:
(by the way, why is ls-R not being used?)

Reply:
Added a comment.

One of the main purposes of the luaotfload font database is that it allows finding fonts by font name, which generaly does not match the file name. So ls-R is not really a good match here.

If I understand correctly, “normal” kpathsea path searching isn't affected by this trick. The subdirectory is only performed when kpathsea does filesystem (i.e. not ls-R-database) searches. This doesn't really happen since all the path components containing references to the nix store in texmf.cnf are prefixed with `!!`,  disabling filesystem searching.